citation-categories/java/src/main/java/org/linqs/psl/examples/citationcategories/Run.java

```diff
@@ -0,0 +1,252 @@
+     */
+    private void runInference() {
+        log.info("Starting inference");
+
+        Partition obsPartition = dataStore.getPartition(PARTITION_EVAL_OBSERVATIONS);
+        Partition targetsPartition = dataStore.getPartition(PARTITION_EVAL_TARGETS);
+        Set<StandardPredicate>set = new HashSet<StandardPredicate>();
+	set.add(model.getStandardPredicate("Link"));
+	Database inferDB = dataStore.getDatabase(targetsPartition, set, obsPartition);
+
+        InferenceApplication inference = new MPEInference(model, inferDB);
+        inference.inference();
+
+        inference.close();
+        inferDB.close();
+
+        log.info("Inference complete");
+    }
+
+    /**
+     * Run statistical evaluation scripts to determine the quality of the inferences
+     * relative to the defined truth.
+     * Note that the target predicate is categorical and we will assign the category with the
+     * highest truth value as true and the rest false.
+     */
+    private void evalResults() {
+
+        // Because the truth data also includes observed data, we will make sure to include the observed
+        // partition here.
+	Set<StandardPredicate>set = new HashSet<StandardPredicate>();
+	set.add(model.getStandardPredicate("Link"));
+        Database resultsDB = dataStore.getDatabase(dataStore.getPartition(PARTITION_EVAL_TARGETS), set, dataStore.getPartition(PARTITION_EVAL_OBSERVATIONS));
+        Database truthDB = dataStore.getDatabase(dataStore.getPartition(PARTITION_EVAL_TRUTH),
+                dataStore.getRegisteredPredicates());
+
+        Evaluator eval = new CategoricalEvaluator(1);
+        eval.compute(resultsDB, truthDB, model.getStandardPredicate("HasCat"));
+        log.info(eval.getAllStats());
+
+        resultsDB.close();
+        truthDB.close();
+    }
+    
+    /**
+     * Writes the output of the model into a file
+     */
+    private void writeOutput() throws IOException {
+        Database resultsDB = dataStore.getDatabase(dataStore.getPartition(PARTITION_EVAL_TARGETS));
+        (new File(OUTPUT_PATH)).mkdirs();
+        FileWriter writer = new FileWriter(Paths.get(OUTPUT_PATH, "HASCAT.txt").toString());
+        for (GroundAtom atom : resultsDB.getAllGroundAtoms(model.getStandardPredicate("HasCat"))) {
+            for (Constant argument : atom.getArguments()) {
+                writer.write(argument.toString() + "\t");
+            }
+	    writer.write("" + atom.getValue() + "\n");
+	}
+	writer.close();
+        resultsDB.close();
+    }
+
+    public void run() {
+	    
+	definePredicates();
+        defineRules();
+        loadData();
+
+        learnWeights();
+        runInference();
+        try {
+	    writeOutput();
+	} catch (IOException ex) {
+	    throw new RuntimeException("Unable to write out results.", ex);
+	}
+        evalResults();
+
+        dataStore.close();
+    }
+
+    /**
+     * Run this model from the command line
+     * @param args - the command line arguments
```

Comment:
Get rid of the `@param` annotation here and in all the Java/Groovy examples.
(I know you weren't the one that put them in, but they seem a bit juvenile.)

Reply:
Clarification: Just for `main()`.
They are not juvenile in general, but the arguments to main are already well documented.